### PR TITLE
Add gutenberg block for reviews

### LINKED ISSUE #2: Add a Gutenberg block to display reviews of a specific plugin
Documentation about creating blocks is at https://developer.wordpress.org/block-editor/tutorials/create-block/

### PR DESCRIPTION
**Pull request by esudalaimadan**
Fix #2

### CODE REVIEW

File: block.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+	"name": "create-block/review-block",
```
Comment (sudar) [inline block.json]: Rename this to `sudar/wp-plugin-review-block`

File: block.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	"title": "Review Block",
+	"category": "widgets",
+	"icon": "smiley",
+	"description": "Example block written with ESNext standard and JSX support – build step required.",
```
Comment (sudar) [inline block.json]: Use the following as description.

`Embed WordPress plugin reviews`.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+	"name": "review-block",
+	"version": "0.1.0",
+	"description": "Example block written with ESNext standard and JSX support – build step required.",
```
Comment (sudar) [inline package.json]: Change the description.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	"author": "The WordPress Contributors",
```
Comment (sudar) [inline package.json]: Change author to `Sudar Muthu`

File: readme.txt [hunk 5]
```diff
@@ -0,0 +1,57 @@
+=== Review Block ===
```
Comment (sudar) [inline readme.txt]: Remove this entire file.

File: reviews-block.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Registers all block assets so that they can be enqueued through the block editor
+ * in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/block-editor/tutorials/block-tutorial/applying-styles-with-stylesheets/
+ */
+class WP_Review_Block {
```
Comment (sudar) [inline reviews-block.php]: Use the name `WPPluginReviewBlock`

File: package.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{
+	"name": "review-block",
```
Comment (sudar) [inline package.json]: Use the name `wp-plugin-review-block`

File: reviews-block.php [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	public function register() {
+		$this->register_block();
+	}
+
+	function register_block() {
```
Comment (sudar) [inline reviews-block.php]: For all methods in the class.

- Add access specifier
- Add phpdoc comment.

File: reviews-block.php [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+		$script_asset_path = "$dir/build/index.asset.php";
+		if ( ! file_exists( $script_asset_path ) ) {
+			throw new Error(
+				'You need to run `npm start` or `npm run build` for the "create-block/review-block" block first.'
+			);
+		}
+		$index_js     = 'build/index.js';
+		$script_asset = require( $script_asset_path );
+		wp_register_script(
+			'create-block-review-block-block-editor',
```
Comment (sudar) [inline reviews-block.php]: Use the name `wp-plugin-review-block-editor`. Do a review and remove reference to `create-block` everywhere.

File: src/edit.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+ * @param {Object} [props]           Properties passed from the editor.
+ * @param {string} [props.className] Class name generated for the block.
+ *
+ * @return {WPElement} Element to render.
+ */
+export default function Edit( { attributes, className, isSelected, setAttributes } ) {
+	return (
+		<div className={ className }>
+            { ! isSelected ? (
+                "Actual review(s) will be shown."
```
Comment (sudar) [inline src/edit.js]: For all strings, pass them to the `__` function, so that they can be translated.

File: src/edit.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+            ) : (
+                <Placeholder label="Reviews Block" instructions="Add plugin name and review count">
+                    <TextControl
+                        label={ __( 'Plugin Name', 'review-block' ) }
```
Comment (sudar) [inline src/edit.js]: Use the correct text domain.

File: src/index.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+ * Internal dependencies
+ */
+import Edit from './edit';
+
+/**
+ * Every block starts by registering a new block type definition.
+ *
+ * @see https://developer.wordpress.org/block-editor/developers/block-api/#registering-a-block
+ */
+registerBlockType( 'create-block/review-block', {
```
Comment (sudar) [inline src/index.js]: Change the namespace, name, description, title etc of the block based on the previews comments.

File: src/index.js [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+	 * Blocks are grouped into categories to help users browse and discover them.
+	 * The categories provided by core are `common`, `embed`, `formatting`, `layout` and `widgets`.
+	 */
+	category: 'widgets',
+
+	/**
+	 * An icon property should be specified to make it easier to identify a block.
+	 * These can be any of WordPress’ Dashicons, or a custom svg element.
+	 */
+	icon: 'smiley',
```
Comment (sudar) [inline src/index.js]: Change this icon to something that represents a review. Search for Dashicons and pick from there.

File: src/save.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
+/**
```
Comment (sudar) [inline src/save.js]: I had asked you to remove this file entirely.

File: reviews-block.php [hunk 15]
```diff
@@ -0,0 +1,60 @@
+<?php
```
Comment (sudar) [inline reviews-block.php]: Rename this file to `wp-plugin-review-block`

**Review by sudar — CHANGES_REQUESTED** [review]

File: block.json (hunk 1 above)
Comment (esudalaimadan) [reply block.json]: Fixed.

File: block.json (hunk 2 above)
Comment (esudalaimadan) [reply block.json]: Fixed.

File: package.json (hunk 3 above)
Comment (esudalaimadan) [reply package.json]: Fixed.

File: package.json (hunk 4 above)
Comment (esudalaimadan) [reply package.json]: Fixed.

File: readme.txt (hunk 5 above)
Comment (esudalaimadan) [reply readme.txt]: Fixed.

File: reviews-block.php (hunk 6 above)
Comment (esudalaimadan) [reply reviews-block.php]: Fixed.

File: package.json (hunk 7 above)
Comment (esudalaimadan) [reply package.json]: Fixed.

File: src/save.js (hunk 14 above)
Comment (esudalaimadan) [reply src/save.js]: Fixed.

File: reviews-block.php (hunk 9 above)
Comment (esudalaimadan) [reply reviews-block.php]: Fixed.

File: src/edit.js (hunk 11 above)
Comment (esudalaimadan) [reply src/edit.js]: Fixed.

File: reviews-block.php (hunk 15 above)
Comment (esudalaimadan) [reply reviews-block.php]: Fixed.

File: src/index.js (hunk 12 above)
Comment (esudalaimadan) [reply src/index.js]: Fixed.

File: src/index.js (hunk 13 above)
Comment (esudalaimadan) [reply src/index.js]: Fixed.

File: reviews-block.php (hunk 8 above)
Comment (esudalaimadan) [reply reviews-block.php]: Fixed.

File: src/edit.js (hunk 10 above)
Comment (esudalaimadan) [reply src/edit.js]: Fixed.

File: wp-plugin-review-block.php [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+	 * Registers review block.
+	 */
+	public function register() {
+		$this->register_block();
+	}
+
+	/**
+	 * Registers review block and its assets.
+	 */
+	public function register_block() {
```
Comment (sudar) [inline wp-plugin-review-block.php]: This function should be `protected`.

**Review by sudar — CHANGES_REQUESTED** [review]

File: wp-plugin-review-block.php (hunk 16 above)
Comment (esudalaimadan) [reply wp-plugin-review-block.php]: Fixed.

**Review by sudar — APPROVED** [review]

**Pull request merged**